Bug: Schema confusion on null in ST_FlipCoordinates

Hitting an optimizer exception in the integration tests for `ST_FlipCoordinates` when testing with `(None, None)`


Exception:
```
E       sedonadb._lib.SedonaError: Optimizer rule 'optimize_projections' failed
E       caused by
E       Check optimizer-specific invariants after optimizer rule: optimize_projections
E       caused by
E       Internal error: Failed due to a difference in schemas: original schema: DFSchema { inner: Schema { fields: [Field { name: "st_flipcoordinates(NULL)", data_type: Null, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }], metadata: {} }, field_qualifiers: [None], functional_dependencies: FunctionalDependencies { deps: [] } }, new schema: DFSchema { inner: Schema { fields: [Field { name: "st_flipcoordinates(NULL)", data_type: Binary, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }], metadata: {} }, field_qualifiers: [None], functional_dependencies: FunctionalDependencies { deps: [] } }.
E       This issue was likely caused by a bug in DataFusion's code. Please help us to resolve this by filing a bug report in our issue tracker: https://github.com/apache/datafusion/issues
```

Note scalars do not have this issue when running `cargo run`:

```
> SELECT ST_FlipCoordinates(NULL);
┌──────────────────────────┐
│ st_flipcoordinates(NULL) │
│         geometry         │
╞══════════════════════════╡
│                          │
└──────────────────────────┘
```

And the unit tests (where the type is clearly defined)  we do not hit this issue.  

Tests are in: https://github.com/apache/sedona-db/blob/main/python/sedonadb/tests/functions/test_functions.py

--- a/python/sedonadb/tests/functions/test_functions.py
+++ b/python/sedonadb/tests/functions/test_functions.py
@@ -294,7 +294,7 @@ def test_st_envelope(eng, geom, expected):
 @pytest.mark.parametrize(
     ("geom", "expected"),
     [
-        # Failing on None for SedonaDB: with datafusion optimizer exception
+        (None, None),
         ("POINT EMPTY", "POINT (nan nan)"),
         ("POLYGON EMPTY", "POLYGON EMPTY"),
         ("LINESTRING EMPTY", "LINESTRING EMPTY"),

